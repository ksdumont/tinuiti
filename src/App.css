* {
  font-family: 'Poppins', Helvetica, Arial, Lucida, sans-serif;
  margin: 0;
  padding: 0;
  text-align: center;
  box-sizing: border-box;
}
html, body {
  background-image: linear-gradient(180deg, rgba(65,120,207,0.5) 0%, rgba(102,0,204, 0.3) 100%);
  height: 100vh;
}
.tinuiti-logo {
  height: 120px;
  width: 250px;
  position: absolute;
  top: 0;
  left: 1rem;
}
/* Search Box Styles */
.search {
  height: 100px;
}
input[type='text'] {
  padding: 10px;
  font-size: 2rem;
  border: 0;
  border-radius: 5px;
  width: 60%;
  margin: auto;
  margin-top: 5rem;
}
input[type=text]:focus{
  outline: 3px solid #4178cf;     
}
/* Product List Styles */
.product-list {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  margin-top: 5rem;
}
/* Card Styles */
.card {
  border: solid 1px #4178cf;
  background-color: white;
  border-radius: 10%;
  height: 225px;
  width: 200px;
  margin: 1rem;
  padding: 1rem;
  box-shadow: 0 5px 10px -5px;
  transition: all 0.3s ease;
  opacity: 0.9;
}
.card:hover {
  box-shadow: -1px -1px 10px 3px #4178cf, 1px 1px 10px 3px #4178cf;
  transform: scale(1.1);
  border-radius: 0;
}
.card-img > img {
  height: 50px;
  width: 90px;
}
.card-body {
  margin: 0.5rem;
  padding: 0.5rem;
}
.star-checked {
  color: #ffa534;
}
.star-empty {
  color: #ffa534;
}
.card-button {
  background-image: linear-gradient(90deg, #4178cf 0%, #6600cc 100%);
  color: white;
  padding: 5px 12px;
  border-radius: 50px;
  font-size: 1.5rem;
  margin-top: 1rem;
  transition: all 0.3s ease;
}
.card-button:hover {
  border-radius: 0;
  transform: scale(1.1);
}

/* product page styles */
.product-page {
padding-top: 7rem;
}
.product-title {
  font-size: 5rem;
}
.product-description {
  color: white;
  margin: 1rem 5rem;
}

.chart-container {
  height: 400px;
  width: 800px;
  padding: 2rem;
  margin: 2rem auto;
  box-shadow: -1px -1px 7px 1px #4178cf, 1px 1px 7px 1px #4178cf;
  background-color: white;
  opacity: 0.8;
}
.loading-text {
  margin-top: 5rem;
}

/* Media Queries */
 @media screen and (max-width: 900px) {
   .search {
     padding-top: 3rem;
   }
   .product-list {
     margin-top: 8rem;
   }
   .chart-container {
    height: 260px;
    width: 470px;
    padding: 1rem;
    margin: 1rem auto;
}
  .product-title {
    font-size: 4rem;
}
  .product-description {
    margin: 1rem 2rem;
} 
 }
 @media screen and (max-width: 500px) {
.tinuiti-logo {
  height: 100px;
  width: 180px;
}
.search {
  height: 80px;
}
input[type='text'] {
  font-size: 1.2rem;
}
.chart-container {
  height: 220px;
  width: 350px;
}
}
@media screen and (max-width: 340px) {
  .chart-container {
  height: 220px;
  width: 300px;
}
}