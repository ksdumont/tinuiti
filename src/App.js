import React, {useState, useEffect} from 'react'
import './App.css';
import MyContext from './MyContext'
import Home from './components/Home'
import ProductPage from './components/ProductPage'
import { Route, Switch } from 'react-router-dom'

function App() {
  const [data, setData] = useState([])

  const fetchData = () => {
    fetch('data.json')
      .then(res => res.json())
      .then(jsonData => setData(jsonData))
  }

  useEffect(() => {
    fetchData()
  }, [])
  
  return (
    <MyContext.Provider value={data}>
        <Switch>
          <Route exact path="/" component={Home} />
          <Route exact path="/product/:id" component={ProductPage} />    
        </Switch>
    </MyContext.Provider>
  );
}

export default App;
