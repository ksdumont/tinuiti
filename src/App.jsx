import React, { useState, useEffect } from 'react';
import './App.css';
import { Route, Switch } from 'react-router-dom';
import ProductContext from './ProductContext';
import Home from './components/Home';
import ProductPage from './components/ProductPage';

function App() {
  const [data, setData] = useState([]);

  const fetchData = () => {
    fetch('data.json')
      .then((res) => res.json())
      .then((jsonData) => setData(jsonData));
  };

  useEffect(() => {
    fetchData();
  }, []);

  return (
    <ProductContext.Provider value={data}>
      <Switch>
        <Route exact path="/" component={Home} />
        <Route exact path="/product/:id" component={ProductPage} />
      </Switch>
    </ProductContext.Provider>
  );
}

export default App;
