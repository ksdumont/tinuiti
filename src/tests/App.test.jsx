import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import App from '../App';

Enzyme.configure({ adapter: new Adapter() });

it('renders correctly', () => {
  const wrapper = shallow(<App />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
