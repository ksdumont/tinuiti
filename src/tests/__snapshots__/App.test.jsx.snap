// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ContextProvider
  value={Array []}
>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/product/:id"
    />
  </Switch>
</ContextProvider>
`;
